### ISSUE TITLE
Mintupgrade to 20 requires for downgrading packages

### ISSUE THREAD

**Issue by 4braincells** [linuxmint/mintupgrade#65]
I am on my way to mint 21, but have 19.3
I already removed unimportant stuff like skype and teams from the repo list,
but I am UNABLE to downgrade mysql as required by mintupgrade.
This blocks my whole work. I need newer glibs bu cant upgrade cause mysql blocks the upgrade.

Some idea how I can trick mintupgrade before I get crazy and do a complete reinstall of the machine (which wil not be Mint then...)

**Comment by 4braincells** [comment]
+ Checking your APT packages...

    - libapache2-mod-php: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - libargon2-0: 0~20190702-0.1+ubuntu18.04.1+deb.sury.org+1, should be 0~20161029-1.1 (from bionic)
    - libfreerdp-client2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libfreerdp2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libgd3: 2.3.3-7+ubuntu18.04.1+deb.sury.org+1, should be 2.2.5-4ubuntu0.5 (from bionic-updates)
    - libidn2-0: 2.3.0-1+ubuntu18.04.1+deb.sury.org+2, should be 2.0.4-1.1ubuntu0.2 (from bionic-updates)
    - libnorm1: 1.5r6+dfsg1-6+ubuntu18.04.1+deb.sury.org+1, should be 1.5r6+dfsg1-6 (from bionic)
    - libpcre2-8-0: 10.40-1+ubuntu18.04.1+deb.sury.org+1, should be 10.31-2 (from bionic)
    - libpcre3: 2:8.44-2+ubuntu18.04.1+deb.sury.org+1, should be 2:8.39-9ubuntu0.1 (from bionic-updates)
    - libpcrecpp0v5: 2:8.44-2+ubuntu18.04.1+deb.sury.org+1, should be 2:8.39-9ubuntu0.1 (from bionic-updates)
    - libsodium23: 1.0.18-1+ubuntu18.04.1+deb.sury.org+1, should be 1.0.16-2 (from bionic)
    - libssl1.1: 1.1.1j-1+ubuntu18.04.1+deb.sury.org+3, should be 1.1.1-1ubuntu2.1~18.04.20 (from bionic-updates)
    - libwinpr2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libxml2: 2.9.14+dfsg-0+ubuntu18.04.1+deb.sury.org+1, should be 2.9.4+dfsg1-6.1ubuntu1.8 (from bionic-updates)
    - libzip4: 1.7.3-1+ubuntu18.04.1+deb.sury.org+2, should be 1.1.2-1.1 (from bionic)
    - libzmq5: 4.3.1-5+ubuntu18.04.1+deb.sury.org+1, should be 4.2.5-1ubuntu0.2 (from bionic-updates)
    - libzstd1: 1.4.8+dfsg-2+ubuntu18.04.1+deb.sury.org+4, should be 1.3.3+dfsg-2ubuntu1.2 (from bionic-updates)
    - mysql-client: 8.0.27-1ubuntu18.04, should be 5.7.41-0ubuntu0.18.04.1 (from bionic-updates)
    - mysql-common: 8.0.27-1ubuntu18.04, should be 5.8+1.0.4 (from bionic)
    - mysql-server: 8.0.27-1ubuntu18.04, should be 5.7.41-0ubuntu0.18.04.1 (from bionic-updates)
    - openssl: 1.1.1j-1+ubuntu18.04.1+deb.sury.org+3, should be 1.1.1-1ubuntu2.1~18.04.20 (from bionic-updates)
    - php-common: 2:93+ubuntu18.04.1+deb.sury.org+2, should be 1:60ubuntu1 (from bionic)
    - php-ldap: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - php-mysql: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - php-xml: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - python3-httplib2: 0.11.3-1+ubuntu18.04.1+deb.sury.org+1, should be 0.9.2+dfsg-1ubuntu0.3 (from bionic-updates)

**Comment by 4braincells** [comment]
Some of these make absolutely no sense, e.g. mysql, cause:
/etc/monit/conf.d$ sudo apt install mysql-server
Reading package lists... Done
Building dependency tree       
Reading state information... Done
mysql-server is already the newest version (8.0.27-1ubuntu18.04).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

the rest might kill other apps, but they are of minor importance  (jenkins, monit, apcupsd, domoticz)

**Comment by carlosmintfan** [comment]
Could you show me the output of `apt policy` and `inxi -r`?

**Comment by 4braincells** [comment]
Apt policy

Package files:
 100 /var/lib/dpkg/status
     release a=now
 500 http://repo.zabbix.com/zabbix/4.4/ubuntu bionic/main i386 Packages
     release o=Zabbix,n=bionic,l=zabbix,c=main,b=i386
     origin repo.zabbix.com
 500 http://repo.zabbix.com/zabbix/4.4/ubuntu bionic/main amd64 Packages
     release o=Zabbix,n=bionic,l=zabbix,c=main,b=amd64
     origin repo.zabbix.com
 500 http://repository.veeam.com/backup/linux/agent/dpkg/debian/public stable/veeam amd64 Packages
     release o=veeam,a=stable,n=stable,l=veeam,c=veeam,b=amd64
     origin repository.veeam.com
 500 https://repo.skype.com/deb stable/main amd64 Packages
     release o=. stable,a=stable,n=stable,l=. stable,c=main,b=amd64
     origin repo.skype.com
 500 http://ppa.launchpad.net/remmina-ppa-team/remmina-next/ubuntu bionic/main i386 Packages
     release v=18.04,o=LP-PPA-remmina-ppa-team-remmina-next,a=bionic,n=bionic,l=remmina-next,c=main,b=i386
     origin ppa.launchpad.net
 500 http://ppa.launchpad.net/remmina-ppa-team/remmina-next/ubuntu bionic/main amd64 Packages
     release v=18.04,o=LP-PPA-remmina-ppa-team-remmina-next,a=bionic,n=bionic,l=remmina-next,c=main,b=amd64
     origin ppa.launchpad.net
 500 https://openhab.jfrog.io/artifactory/openhab-linuxpkg stable/main i386 Packages
     release o=Artifactory,a=stable,n=stable,l=Artifactory,c=main,b=i386
     origin openhab.jfrog.io
 500 https://openhab.jfrog.io/artifactory/openhab-linuxpkg stable/main amd64 Packages
     release o=Artifactory,a=stable,n=stable,l=Artifactory,c=main,b=amd64
     origin openhab.jfrog.io
 500 http://archive.canonical.com/ubuntu bionic/partner i386 Packages
     release v=18.04,o=Canonical,a=bionic,n=bionic,l=Partner archive,c=partner,b=i386
     origin archive.canonical.com
 500 http://archive.canonical.com/ubuntu bionic/partner amd64 Packages
     release v=18.04,o=Canonical,a=bionic,n=bionic,l=Partner archive,c=partner,b=amd64
     origin archive.canonical.com
 500 http://security.ubuntu.com/ubuntu bionic-security/multiverse i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=multiverse,b=i386
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=multiverse,b=amd64
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/universe i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=universe,b=i386
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=universe,b=amd64
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/restricted i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=restricted,b=i386
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=restricted,b=amd64
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/main i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=main,b=i386
     origin security.ubuntu.com
 500 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-security,n=bionic,l=Ubuntu,c=main,b=amd64
     origin security.ubuntu.com
 100 http://archive.ubuntu.com/ubuntu bionic-backports/universe i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-backports,n=bionic,l=Ubuntu,c=universe,b=i386
     origin archive.ubuntu.com
 100 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-backports,n=bionic,l=Ubuntu,c=universe,b=amd64
     origin archive.ubuntu.com
 100 http://archive.ubuntu.com/ubuntu bionic-backports/main i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-backports,n=bionic,l=Ubuntu,c=main,b=i386
     origin archive.ubuntu.com
 100 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-backports,n=bionic,l=Ubuntu,c=main,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=multiverse,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=multiverse,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/universe i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=universe,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=universe,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/restricted i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=restricted,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=restricted,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/main i386 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=main,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic-updates,n=bionic,l=Ubuntu,c=main,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/multiverse i386 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=multiverse,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=multiverse,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/universe i386 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=universe,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=universe,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/restricted i386 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=restricted,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=restricted,b=amd64
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/main i386 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=main,b=i386
     origin archive.ubuntu.com
 500 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages
     release v=18.04,o=Ubuntu,a=bionic,n=bionic,l=Ubuntu,c=main,b=amd64
     origin archive.ubuntu.com
 500 https://mirror.dogado.de/linuxmint tricia/backport i386 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=backport,b=i386
     origin mirror.dogado.de
 500 https://mirror.dogado.de/linuxmint tricia/backport amd64 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=backport,b=amd64
     origin mirror.dogado.de
 500 https://mirror.dogado.de/linuxmint tricia/import i386 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=import,b=i386
     origin mirror.dogado.de
 500 https://mirror.dogado.de/linuxmint tricia/import amd64 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=import,b=amd64
     origin mirror.dogado.de
 700 https://mirror.dogado.de/linuxmint tricia/upstream i386 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=upstream,b=i386
     origin mirror.dogado.de
 700 https://mirror.dogado.de/linuxmint tricia/upstream amd64 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=upstream,b=amd64
     origin mirror.dogado.de
 500 https://mirror.dogado.de/linuxmint tricia/main i386 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=main,b=i386
     origin mirror.dogado.de
 500 https://mirror.dogado.de/linuxmint tricia/main amd64 Packages
     release v=19.3,o=linuxmint,a=tricia,n=tricia,l=linuxmint,c=main,b=amd64
     origin mirror.dogado.de
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-tools i386 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-tools,b=i386
     origin repo.mysql.com
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-tools amd64 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-tools,b=amd64
     origin repo.mysql.com
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-8.0 i386 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-8.0,b=i386
     origin repo.mysql.com
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-8.0 amd64 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-8.0,b=amd64
     origin repo.mysql.com
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-apt-config i386 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-apt-config,b=i386
     origin repo.mysql.com
 500 http://repo.mysql.com/apt/ubuntu bionic/mysql-apt-config amd64 Packages
     release o=MySQL,n=bionic,l=MySQL,c=mysql-apt-config,b=amd64
     origin repo.mysql.com
 500 https://pkg.jenkins.io/debian-stable binary/ Packages
     release o=jenkins.io,a=binary,c=
     origin pkg.jenkins.io
 500 http://dl.google.com/linux/chrome/deb stable/main amd64 Packages
     release v=1.0,o=Google LLC,a=stable,n=stable,l=Google,c=main,b=amd64
     origin dl.google.com
 500 http://apt.llvm.org/focal llvm-toolchain-focal-12/main i386 Packages
     release n=llvm-toolchain-focal-12,c=main,b=i386
     origin apt.llvm.org
 500 http://apt.llvm.org/focal llvm-toolchain-focal-12/main amd64 Packages
     release n=llvm-toolchain-focal-12,c=main,b=amd64
     origin apt.llvm.org
Pinned packages:

**Comment by 4braincells** [comment]
inxi -r
Repos:     No active apt repos in: /etc/apt/sources.list 
           Active apt repos in: /etc/apt/sources.list.d/additional-repositories.list 
           1: deb http://apt.llvm.org/focal/ llvm-toolchain-focal-12 main
           Active apt repos in: /etc/apt/sources.list.d/gitlab_gitlab-ce.list 
           1: deb https://packages.gitlab.com/gitlab/gitlab-ce/linuxmint/ tricia main
           2: deb-src https://packages.gitlab.com/gitlab/gitlab-ce/linuxmint/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/gitlab_gitlab-ee.list 
           1: deb https://packages.gitlab.com/gitlab/gitlab-ee/linuxmint/ tricia main
           2: deb-src https://packages.gitlab.com/gitlab/gitlab-ee/linuxmint/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/google-chrome.list 
           1: deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/jenkins.list 
           1: deb https://pkg.jenkins.io/debian-stable binary/
           Active apt repos in: /etc/apt/sources.list.d/mysql.list 
           1: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-apt-config
           2: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-8.0
           3: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-tools
           4: deb-src http://repo.mysql.com/apt/ubuntu/ bionic mysql-8.0
           Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list 
           1: deb https://mirror.dogado.de/linuxmint tricia main upstream import backport
           2: deb http://archive.ubuntu.com/ubuntu bionic main restricted universe multiverse
           3: deb http://archive.ubuntu.com/ubuntu bionic-updates main restricted universe multiverse
           4: deb http://archive.ubuntu.com/ubuntu bionic-backports main restricted universe multiverse
           5: deb http://security.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse
           6: deb http://archive.canonical.com/ubuntu/ bionic partner
           Active apt repos in: /etc/apt/sources.list.d/ondrej-php-bionic.list 
           1: deb http://ppa.launchpad.net/ondrej/php/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/openhab.list 
           1: deb [signed-by=/usr/share/keyrings/openhab.gpg] https://openhab.jfrog.io/artifactory/openhab-linuxpkg stable main
           Active apt repos in: /etc/apt/sources.list.d/pgdg.list 
           1: deb http://apt.postgresql.org/pub/repos/apt tricia-pgdg main
           Active apt repos in: /etc/apt/sources.list.d/pgsql.list 
           1: deb http://apt.postgresql.org/pub/repos/apt/ tricia-pgdg main 15
           Active apt repos in: /etc/apt/sources.list.d/php.list 
           1: deb https://packages.sury.org/php/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/remmina-ppa-team-remmina-next-bionic.list 
           1: deb http://ppa.launchpad.net/remmina-ppa-team/remmina-next/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/skype-stable.list 
           1: deb [arch=amd64] https://repo.skype.com/deb stable main
           Active apt repos in: /etc/apt/sources.list.d/teams.list 
           1: deb [arch=amd64] https://packages.microsoft.com/repos/ms-teams stable main
           Active apt repos in: /etc/apt/sources.list.d/veeam.list 
           1: deb [arch=amd64] http://repository.veeam.com/backup/linux/agent/dpkg/debian/public stable veeam
           Active apt repos in: /etc/apt/sources.list.d/zabbix.list 
           1: deb http://repo.zabbix.com/zabbix/4.4/ubuntu bionic main
           2: deb-src http://repo.zabbix.com/zabbix/4.4/ubuntu bionic main

**Comment by 4braincells** [comment]
inxi -r
Repos:     No active apt repos in: /etc/apt/sources.list 
           Active apt repos in: /etc/apt/sources.list.d/additional-repositories.list 
           1: deb http://apt.llvm.org/focal/ llvm-toolchain-focal-12 main
           Active apt repos in: /etc/apt/sources.list.d/gitlab_gitlab-ce.list 
           1: deb https://packages.gitlab.com/gitlab/gitlab-ce/linuxmint/ tricia main
           2: deb-src https://packages.gitlab.com/gitlab/gitlab-ce/linuxmint/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/gitlab_gitlab-ee.list 
           1: deb https://packages.gitlab.com/gitlab/gitlab-ee/linuxmint/ tricia main
           2: deb-src https://packages.gitlab.com/gitlab/gitlab-ee/linuxmint/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/google-chrome.list 
           1: deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main
           Active apt repos in: /etc/apt/sources.list.d/jenkins.list 
           1: deb https://pkg.jenkins.io/debian-stable binary/
           Active apt repos in: /etc/apt/sources.list.d/mysql.list 
           1: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-apt-config
           2: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-8.0
           3: deb http://repo.mysql.com/apt/ubuntu/ bionic mysql-tools
           4: deb-src http://repo.mysql.com/apt/ubuntu/ bionic mysql-8.0
           Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list 
           1: deb https://mirror.dogado.de/linuxmint tricia main upstream import backport
           2: deb http://archive.ubuntu.com/ubuntu bionic main restricted universe multiverse
           3: deb http://archive.ubuntu.com/ubuntu bionic-updates main restricted universe multiverse
           4: deb http://archive.ubuntu.com/ubuntu bionic-backports main restricted universe multiverse
           5: deb http://security.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse
           6: deb http://archive.canonical.com/ubuntu/ bionic partner
           Active apt repos in: /etc/apt/sources.list.d/ondrej-php-bionic.list 
           1: deb http://ppa.launchpad.net/ondrej/php/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/openhab.list 
           1: deb [signed-by=/usr/share/keyrings/openhab.gpg] https://openhab.jfrog.io/artifactory/openhab-linuxpkg stable main
           Active apt repos in: /etc/apt/sources.list.d/pgdg.list 
           1: deb http://apt.postgresql.org/pub/repos/apt tricia-pgdg main
           Active apt repos in: /etc/apt/sources.list.d/pgsql.list 
           1: deb http://apt.postgresql.org/pub/repos/apt/ tricia-pgdg main 15
           Active apt repos in: /etc/apt/sources.list.d/php.list 
           1: deb https://packages.sury.org/php/ tricia main
           Active apt repos in: /etc/apt/sources.list.d/remmina-ppa-team-remmina-next-bionic.list 
           1: deb http://ppa.launchpad.net/remmina-ppa-team/remmina-next/ubuntu bionic main
           Active apt repos in: /etc/apt/sources.list.d/skype-stable.list 
           1: deb [arch=amd64] https://repo.skype.com/deb stable main
           Active apt repos in: /etc/apt/sources.list.d/teams.list 
           1: deb [arch=amd64] https://packages.microsoft.com/repos/ms-teams stable main
           Active apt repos in: /etc/apt/sources.list.d/veeam.list 
           1: deb [arch=amd64] http://repository.veeam.com/backup/linux/agent/dpkg/debian/public stable veeam
           Active apt repos in: /etc/apt/sources.list.d/zabbix.list 
           1: deb http://repo.zabbix.com/zabbix/4.4/ubuntu bionic main
           2: deb-src http://repo.zabbix.com/zabbix/4.4/ubuntu bionic main

**Comment by 4braincells** [comment]
> Could you show me the output of `apt policy` and `inxi -r`?

didnt get noticed of this comment... :-/

**Comment by carlosmintfan** [comment]
Okay, I see you have many 3rd party repos which could block the upgrade. At which stage does mintupgrade fail? Int the "Phase 1: Introduction" or after the second?

**Comment by carlosmintfan** [comment]
Don't go crazy, I'll try to help you best ;)

**Comment by 4braincells** [comment]
> Don't go crazy, I'll try to help you best ;)
I won't go crazy, I'm too old for that :-p

This "can" is from the time when it was my secondary desktop, so most of the shmodder can be deleted.

I dont care for the Microsoft/Citrix/remmina/skype/google BS on this machine,
zabbix is already deleted. openhab stuff can also go ... 
dont know what sury is.
veeam doesnt work anyway, so I maybe delete it anyway and reinstall

What has to remain untouched should be mysql, postgres, jenkins, monit, bind9.

just wonder who brought mongo to here :-D

so, except mysql, everything can be wiped.

mintupgrade check output:

 + Checking your Linux Mint codename...
  + Checking your Linux Mint edition...
  + Checking your Timeshift configuration...
  + Updating cache...
...
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list.d/pgdg.list:1 and /etc/apt/sources.list.d/pgsql.list:1

  + Checking your APT repositories...

------------------------------------------------
!!  ERROR: The following 3rd party repositories were detected. Disable them and refresh your APT cache.

    - . stable (. stable) - repo.skype.com
    - Artifactory (Artifactory) - openhab.jfrog.io
    - Google (Google LLC) - dl.google.com
    - MySQL (MySQL) - repo.mysql.com
    - remmina-next (LP-PPA-remmina-ppa-team-remmina-next) - ppa.launchpad.net
    - veeam (veeam) - repository.veeam.com
    - zabbix (Zabbix) - repo.zabbix.com

!!  Exiting.
------------------------------------------------

  + Restoring your backed up APT sources...

**Comment by carlosmintfan** [comment]
Ah, I overlooked you're using the old mintupgrade tool (as required for the 19.3 > 20 upgrade, you'll get a graphical one for 20.3 > 21)
But the error in the "Checking your APT *packages*" phase - did you get that after - bypassing the "Checking your APT *repositories*" phase??

**Comment by carlosmintfan** [comment]
As for the error with disabling the 3rd party repos, I could send you a hack tomorrow how to bypass that (if you haven't already done that).
Another solution could be backing up the database files (maybe not really necessary, but to be sure), then disabling and removing the 3rd party mysql etc. repos, downgrading/removing the packages in mintsources under the Maitenance page and reinstalling everything after the upgrade…

**Comment by 4braincells** [comment]
AFTER REMOVING mysql.list temporarily, following package problems remain:

 - libapache2-mod-php: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - libargon2-0: 0~20190702-0.1+ubuntu18.04.1+deb.sury.org+1, should be 0~20161029-1.1 (from bionic)
    - libfreerdp-client2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libfreerdp2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libgd3: 2.3.3-9+ubuntu18.04.1+deb.sury.org+1, should be 2.2.5-4ubuntu0.5 (from bionic-updates)
    - libidn2-0: 2.3.0-1+ubuntu18.04.1+deb.sury.org+2, should be 2.0.4-1.1ubuntu0.2 (from bionic-updates)
    - libnorm1: 1.5r6+dfsg1-6+ubuntu18.04.1+deb.sury.org+1, should be 1.5r6+dfsg1-6 (from bionic)
    - libpcre2-8-0: 10.40-1+ubuntu18.04.1+deb.sury.org+1, should be 10.31-2 (from bionic)
    - libpcre3: 2:8.44-2+ubuntu18.04.1+deb.sury.org+1, should be 2:8.39-9ubuntu0.1 (from bionic-updates)
    - libsodium23: 1.0.18-1+ubuntu18.04.1+deb.sury.org+1, should be 1.0.16-2 (from bionic)
    - libssl1.1: 1.1.1j-1+ubuntu18.04.1+deb.sury.org+3, should be 1.1.1-1ubuntu2.1~18.04.23 (from bionic-updates)
    - libwinpr2-2: 2.7.0+git202204251304-0+remmina202205020710.r40ee5d3bc.da996883ab~ubuntu18.04.1, should be 2.2.0+dfsg1-0ubuntu0.18.04.4 (from bionic-updates)
    - libxml2: 2.9.14+dfsg-0.1+ubuntu18.04.1+deb.sury.org+1, should be 2.9.4+dfsg1-6.1ubuntu1.9 (from bionic-updates)
    - libzip4: 1.7.3-1+ubuntu18.04.1+deb.sury.org+2, should be 1.1.2-1.1 (from bionic)
    - libzmq5: 4.3.1-5+ubuntu18.04.1+deb.sury.org+1, should be 4.2.5-1ubuntu0.2 (from bionic-updates)
    - libzstd1: 1.4.8+dfsg-2+ubuntu18.04.1+deb.sury.org+4, should be 1.3.3+dfsg-2ubuntu1.2 (from bionic-updates)
    - mysql-client: 8.0.27-1ubuntu18.04, should be 5.7.42-0ubuntu0.18.04.1 (from bionic-updates)
    - mysql-common: 8.0.27-1ubuntu18.04, should be 5.8+1.0.4 (from bionic)
    - mysql-server: 8.0.27-1ubuntu18.04, should be 5.7.42-0ubuntu0.18.04.1 (from bionic-updates)
    - openssl: 1.1.1j-1+ubuntu18.04.1+deb.sury.org+3, should be 1.1.1-1ubuntu2.1~18.04.23 (from bionic-updates)
    - php-common: 2:93+ubuntu18.04.1+deb.sury.org+2, should be 1:60ubuntu1 (from bionic)
    - php-ldap: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - php-mysql: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - php-xml: 2:8.2+93+ubuntu18.04.1+deb.sury.org+2, should be 1:7.2+60ubuntu1 (from bionic)
    - python3-httplib2: 0.11.3-1+ubuntu18.04.1+deb.sury.org+1, should be 0.9.2+dfsg-1ubuntu0.3 (from bionic-updates)

I do not think that it makes sense to clean that mess up.
So I have a tendency to make  a clean installation.

Just have to decide now if it gets mint, ubuntu or naked debian.

**Comment by Jeremy7701** [comment]
Suggest LMDE6 as this is Debian clothed in Cinnamon.

**Comment by 4braincells** [comment]
> Suggest LMDE6 as this is Debian clothed in Cinnamon.

hm. this is an old i5 quadcore/16GB box which does its job.
Despite I used it for citrix sessions with a customer, it ihas been and is mainly a build server.
I had decided for Mint because of easy installation.

But see the mambojumbo I have to do to migrate 18.4 to 21 ... only because 18.4 is "incompatible" with mysql 8.
I mean this is a self-made and rather superfluous dependency with no reason except to support DAUs.
I need the freedom to install any package in any version and not be fined if I want to upgrade the system without reinstall

So, I have a tendency to simply use naked Debian + xfce. For the rest, APT and my remaining 4 brain cells must do the job :-D

But I am open for arguments ...

**Comment by carlosmintfan** [comment]
Well, that's all upstream stuff, so it shouldn't be mint that is to blame ;) Usign newer package versions from 3rd party repos of packages that exist (as older versions) in the standard debian/ubuntu repos are a typical cause for upgrade failure. It shouldn't be very hard to clean that mess up:
1. Backup all important databases in /var/lib of the packages you're going to downgrade
2. Disable ALL 3rd party package sources
3. Open mintsources (the software sources tool)
4. Go to maintenance in the sidebar
5. Click on 'Downgrade foreign packages'
6. Downgrade everything
7. If that works, you can upgrade

**Comment by carlosmintfan** [comment]
And if you decide to upgrade, I'd recommend you to restore your 3rd party repos only after you've done all upgrade steps (to 20, to 20.3, to 21 and, if you want, to 21.3) (and appropiately change the codename 'tricia' in them to the last release you're upgrading to).
